modules/post/windows/gather/credentials/securecrt.rb

```diff
@@ -73,13 +73,19 @@ def enum_session_file(path)
       end
 
       file = try_encode_file(file_contents)
+      protocol = Regexp.compile('S:"Protocol Name"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
       hostname = Regexp.compile('S:"Hostname"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
       password = Regexp.compile('S:"Password"=u([0-9a-f]+)').match(file) ? securecrt_crypto(Regexp.last_match(1)) : nil
       passwordv2 = Regexp.compile('S:"Password V2"=02:([0-9a-f]+)').match(file) ? securecrt_crypto_v2(Regexp.last_match(1)) : nil
-      port = Regexp.compile('D:"\[SSH2\] Port"=([0-9a-f]{8})').match(file) ? Regexp.last_match(1).to_i(16).to_s : nil
+      port = Regexp.compile("D:\"\\\[#{protocol}\\\] Port\"=([0-9a-f]{8})").match(file) ? Regexp.last_match(1).to_i(16).to_s : nil
+      port = Regexp.compile('D:"Port"=([0-9a-f]{8})').match(file) ? Regexp.last_match(1).to_i(16).to_s : nil if !port
       username = Regexp.compile('S:"Username"=([^\r\n]*)').match(file) ? Regexp.last_match(1) : nil
+
+      next unless (hostname && port && protocol)
```

Comment:
So this in my eyes is not the right way to do things. The idea here should be that we still add elements to `tbl` here even if they are `nil`. The place where the `nil` checks should occur is before calling `securecrt_store_config`. This will allow us to still print and store the contents of the `tbl` table, even if it has `nil` elements, whilst also ensuring that `securecrt_store_config`, which will store the credentials into the credential database, will only be called if we have the following (and none of them are set to `nil`):

- username
- password
- hostname
- protocol
- port

I'll see if I can't help fix this in my next set of commits.

Reply:
This was fixed in the last commit and the new check adjustments that I will upload shortly should round off the final touches on fixes for this issue. Marking this as resolved for now.